pkg/resource/logging-secret/reconciler.go

```diff
@@ -44,7 +44,17 @@ func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Inter
 	}
 
 	// Get desired secret
-	desiredLoggingSecret, err := GenerateLoggingSecret(lc, &loggingCredentialsSecret, lokiURL)
+	// Get desired config
+	var desiredLoggingSecret v1.Secret
+	switch lc.GetLoggingAgent() {
+	case "promtail":
+		desiredLoggingSecret, err = GeneratePromtailLoggingSecret(lc, &loggingCredentialsSecret, lokiURL)
```

Comment:
I think we should add an interface that generates the logging secret and the deciding code will be there and not in the reconciler because this is missing the app version (if the observability bundle does not support alloy then that's not working at all)

Reply:
Can you elaborate on what this interface would look like ?

Reply:
I think i would have kept the logging-secret.go resource and have the logging agent flag done there. 

Like that you can share the secret creation and only the content changes?

Reply:
I moved the secret creation and logic to select logging agent to `GenerateLoggingSecret`